lib/rack/lint.rb

```diff
@@ -18,6 +18,33 @@ class Lint
     REQUEST_PATH_AUTHORITY_FORM = /\A[^\/:]+:\d+\z/
     REQUEST_PATH_ASTERISK_FORM = '*'
 
+    # Match a host name, according to RFC3986.
+    # Copied from `URI::RFC3986_Parser::HOST` because older Ruby versions (< 3.3) don't expose it.
+    HOST_PATTERN = /
+      (?<IP-literal>\[(?:
```

Comment:
I think if we are copying it, we should drop the unused named captures.

Reply:
I think it serves as documentation. With `match?` it won't matter anyway, and it avoids any effort to copy an updated Regexp in the future (from `uri`).

Reply:
Fair enough.  Let's make both constants private, though.

Reply:
That makes sense. Maybe we should do that for the other constants too.